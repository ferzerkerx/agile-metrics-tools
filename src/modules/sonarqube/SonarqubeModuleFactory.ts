import { CollectorModuleFactory, Utils } from '../../metrics';
import { SonarCollectorConfig, SonarMetricItem } from './collector/Types';
import { SonarService } from './Types';
import { SonarClientImpl } from './SonarClientImpl';
import { SonarRepositoryImpl } from './SonarRepositoryImpl';
import { SonarServiceImpl } from './SonarServiceImpl';
import { SonarCollectorsService } from './collector/SonarCollectorsService';

export class SonarqubeModuleFactory
  implements CollectorModuleFactory<SonarCollectorConfig, SonarMetricItem> {
  private static sonarService(): SonarService {
    Utils.checkEnvVar('SONAR_HOST');

    const sonarClient = new SonarClientImpl({
      host: `${process.env.SONAR_HOST}`,
    });
    const sonarRepository = new SonarRepositoryImpl(sonarClient);
    return new SonarServiceImpl(sonarRepository);
  }
  collectorInstance(): SonarCollectorsService {
    return new SonarCollectorsService(SonarqubeModuleFactory.sonarService());
  }

  collectorConfiguration(obj: any): SonarCollectorConfig {
    return new SonarCollectorConfig(obj);
  }
}
