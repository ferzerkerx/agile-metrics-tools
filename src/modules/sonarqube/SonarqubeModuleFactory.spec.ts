import { SonarqubeModuleFactory } from "./SonarqubeModuleFactory";

describe("SonarqubeModuleFactory", () => {
  const sonarqubeModuleFactory: SonarqubeModuleFactory = new SonarqubeModuleFactory();

  beforeEach(() => {
    jest.resetModules();
    delete process.env.SONAR_HOST;
  });

  it("should create collector", () => {
    process.env.SONAR_HOST = "SOME_TOKEN";
    const collectorInstance = sonarqubeModuleFactory.collectorInstance();
    expect(collectorInstance).not.toBeNull();
  });

  it("should fail to create collector", () => {
    expect(
      sonarqubeModuleFactory.collectorInstance
    ).toThrowErrorMatchingSnapshot();
  });

  it("should create collectorConfiguration", () => {
    const collectorConfiguration = sonarqubeModuleFactory.collectorConfiguration(
      {}
    );
    expect(collectorConfiguration).not.toBeNull();
  });
});
