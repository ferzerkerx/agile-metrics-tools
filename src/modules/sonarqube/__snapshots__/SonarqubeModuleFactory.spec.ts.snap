// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SonarqubeModuleFactory should fail to create collector 1`] = `"env.SONAR_HOST not set!"`;
